selftests/unit/test_nrunner.py

```diff
@@ -220,6 +224,63 @@ def test_runner_python_unittest(self):
         self.assertTrue(result['output'].startswith(output1))
         self.assertTrue(result['output'].endswith(output2))
 
+    def test_runner_tap_fail(self):
+        tap_script = """#!/bin/bash
+echo '1..2'
+echo '# Defining an basic test'
+echo 'ok 1 - description 1'
+echo 'not ok 2 - description 2'"""
+        tap_path = os.path.join(self.tmpdir.name, 'tap.sh')
+
+        with open(tap_path, 'w') as fp:
+            fp.write(tap_script)
+
+        runnable = nrunner.Runnable('tap', '/bin/bash', tap_path)
```

Comment:
A busybox-like `/bin/sh`, non-bash, should be enough to run this script, and improves portability.  It's also the choice of the majority of tests that use a shell (and we should switch the few ones that don't need bash but use it).